pkcs11/_pkcs11.pyx

```diff
@@ -268,36 +268,37 @@ class Token(types.Token):
             raise ArgumentsBad("Set either `user_pin` or `so_pin`")
         elif user_pin is PROTECTED_AUTH:
             pin = None
-            user_type = CKU_USER
+            user_type = user_type if user_type else CKU_USER
         elif so_pin is PROTECTED_AUTH:
             pin = None
-            user_type = CKU_SO
+            user_type = user_type if user_type else CKU_USER
         elif user_pin is not None:
             pin = user_pin.encode('utf-8')
-            user_type = CKU_USER
+            user_type = user_type if user_type else CKU_USER
```

Comment:
Avoid truthy statements, as 0 is a valid user type value. Compare directly to None.